File: backend/app/models/review.rb
class Review < ApplicationRecord
  include ActionView::Helpers::DateHelper
  belongs_to :product
  belongs_to :user
  validates :title, :content, :user_id, :product_id, presence: true
  validates :rate, numericality: {
    greater_than_or_equal_to: 1,
    less_than_or_equal_to: 5
  }, presence: true

  counter_culture :product
  counter_culture :product, column_name: 'rate_sum', delta_column: 'rate'

  scope :get_reviews, ->(id) { where(product_id: id) }

  def self.api_json(product_id)
    reviews = Review.get_reviews(product_id)
    reviews.map(&:as_json)
  end

  def as_json
    {
      id: id,
      title: title,
      content: content,
      rate: rate,
      time_ago: time_ago_in_words(created_at),
      user_id: user_id
    }
  end
end

File: frontend/src/mock/Review.ts
import { Review } from 'model/index';

export const reviews: Review[] = [
  {
    id: 1,
    title: 'レビューのタイトル１',
    content: 'レビューの内容１',
    rate: 5,
    user_id: 1,
    time_ago: new Date(),
  },
  {
    id: 2,
    title: 'レビューのタイトル2',
    content: 'レビューの内容2',
    rate: 4,
    user_id: 2,
    time_ago: new Date(),
  },
];

export default reviews;
